pandas/_libs/tslib.pyx

```diff
@@ -726,16 +728,14 @@ class Timestamp(_Timestamp):
             _tzinfo = tzinfo
 
         # reconstruct & check bounds
-        value = pandas_datetimestruct_to_datetime(PANDAS_FR_ns, &dts)
+        ts_input = datetime(dts.year, dts.month, dts.day, dts.hour, dts.min,
```

Comment:
Is there a fast low-level function that takes a year, month, day, hour, minute, second, nano and a tzinfo and calculates value so that I can avoid creating a `datetime` object? Something like `convert_to_tsobject()` but with a different signature? Or can I call this function in a different way?